packages/react-components/src/shared/src/useForwardInputApi.ts

```diff
@@ -8,14 +10,14 @@ const API_METHODS = [
     "checkValidity",
     "reportValidity",
     "setCustomValidity"
-];
+] as const;
 
-export function useForwardInputApi(inputRef) {
-    return targetRef => {
+export function useForwardInputApi<T extends Record<string, any>>(inputRef: MutableRefObject<T | undefined>) {
```

Comment:
Does T is really a Record? The ref will always be bound to a ReactElement so it should always be an HtmlElement

Reply:
HTMLInputElement is perfect, thank you